pkg/hs/urbit-noun/lib/Urbit/Noun/Conversions.hs

```diff
@@ -552,6 +555,18 @@ instance FromNoun Knot where
       then pure (MkKnot txt)
       else fail ("Non-ASCII chars in knot: " <> unpack txt)
 
+-- equivalent of (cury scot %t)
+textAsTa :: Text -> Text
+textAsTa = ("~~" <>) . concatMap \case
+  ' ' -> "."
+  '.' -> "~."
+  '~' -> "~~"
+  c   ->
+    if C.isAlphaNum c || (c == '-') then
+      T.singleton c
+    else
+      if C.ord c < 0x10 then "~0" else "~"
+      <> (pack $ N.showHex (C.ord c) ".")
```

Comment:
Okay this is definitely not performant. This is a case where we should be using Text Builders.

Reply:
As discussed out of band, since this is the `concatMap` from `Data.Text`, this is O(n), so perfectly fine.